integration/cluster_api_endpoints_test.go

```diff
@@ -0,0 +1,220 @@
+// +build integration
+
+package integration_test
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	awseks "github.com/aws/aws-sdk-go/service/eks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	. "github.com/weaveworks/eksctl/integration/matchers"
+	. "github.com/weaveworks/eksctl/integration/runner"
+
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+)
+
+const (
+	createCluster    = `CreateCluster`
+	updateCluster    = `UpdateCluster`
+	endpointPubTmpl  = `EndpointPublicAccess: %v`
+	endpointPrivTmpl = `EndpointPrivateAccess: %v`
+)
+
+var False = false
+
+func setEndpointConfig(cfg *api.ClusterConfig, privateAccess, publicAccess bool) {
+	cfg.VPC.ClusterEndpoints.PrivateAccess = &privateAccess
+	cfg.VPC.ClusterEndpoints.PublicAccess = &publicAccess
+}
+
+func generateName(prefix string) string {
+	if clusterName == "" {
+		clusterName = cmdutils.ClusterName("", "")
+	}
+	return fmt.Sprintf("%v-%v", prefix, clusterName)
+}
+
+func setMetadata(cfg *api.ClusterConfig, name, region string) {
+	cfg.Metadata.Name = name
+	cfg.Metadata.Region = region
+}
+
+func printSeparator(op, clname string) {
```

Comment:
It doesn't work with the standard BeforeEach or JustBeforeEach functions, because they can only reside within a Describe, Context, or It block.  DescribeTable only allows the fun(x entryStruct) and the entry's themselves.  Putting it before DescribeTables (under the main Describe block opening) doesn't work because the Entries are passed in only to the DescribeTable function.  So putting it in that function is about as close as you can get to using the Ginkgo provided tools.

Reply:
I did end up making 'deleteCluster' another endpointAccessCase.Type value though and did a little refactoring to the test as a result.   Pushing that shortly.

Reply:
I have updated this so that the printSeparator call happens at the beginning of each test case and passes the 'op' from the use case definition.